spec/defines/snippet_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'spec_helper'
+
+describe 'bird::snippet', type: :define do
+  on_supported_os.each do |os, facts|
+    context "on #{os} " do
+      let :facts do
+        facts
+      end
```

Comment:
I've gotten in the habit of using `os_facts` to avoid the variable overlap and write it a bit more short hand:
```suggestion
  on_supported_os.each do |os, os_facts|
    context "on #{os}" do
      let(:facts) { os_facts }
```